Plugins/GeoModel/src/SplitModulesByRadius.cpp

```diff
@@ -0,0 +1,84 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/GeoModel/SplitModulesByRadius.hpp"
+
+#include "Acts/Surfaces/AnnulusBounds.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <iostream>
+
+namespace Acts {
+
+std::optional<std::vector<std::pair<std::shared_ptr<Surface>,
+                                    std::shared_ptr<GeoModelDetectorElement>>>>
+ModuleByRadiusSplitter::split(
```

Comment:
Does it make sense to add a test for this function?

Reply:
Yeah actually it would, I wanted to do this anyways after sonarcloud complained.